### PR TITLE
Fix Issue 21275 - Overload resolution bypasses private access

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by RazvanN7** [comment]
What shoud I do about this:

```
source/contentprovider.d(330): [unittest] Source code line length exceeds 48 limit in 'Lubeck': import mir.ndslice: magic, repeat, as, slice, byDim;
--
  | object.Exception@source/contentprovider.d(330): Source code line length exceeds 48 limit in 'Lubeck': import mir.ndslice: magic, repeat, as, slice, byDim;
```

**Comment by thewilsonator** [comment]
Ignore it and cc @9il to fix it and tag another release, then update the CI scripts to the new version.

**Comment by Geod24** [comment]
This pull request introduced a regression: https://issues.dlang.org/show_bug.cgi?id=21670
Which is preventing Ocean from using a more recent version of DMD.

**Comment by ibuclaw** [comment]
This PR caused a regression.

https://issues.dlang.org/show_bug.cgi?id=23592

File: changelog/deprecate_visibility.dd [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  auto f = Foo();
+  f.x = 3;              // ok to call private method
+  writeln(f);
+}
+===========================
+```
+
+Starting from this version a deprecation message will be issued
+in such situations.
```
Comment (ibuclaw) [inline changelog/deprecate_visibility.dd:47]: ... except when you add parentheses `f.x() = 3` :facepalm:

File: src/dmd/expressionsem.d [hunk 2]
```diff
@@ -1273,6 +1273,15 @@ private Expression resolvePropertiesX(Scope* sc, Expression e1, Expression e2 =
             {
                 if (fd.errors)
                     return ErrorExp.get();
+                if (!checkSymbolAccess(sc, fd))
+                {
+                    // @@@DEPRECATED_2020-10@@@
+                    // When turning into error, uncomment the return statement
+                    TypeFunction tf = cast(TypeFunction)fd.type;
+                    deprecation(loc, "Function `%s` of type `%s` is not accessible from module `%s`",
```
Comment (ibuclaw) [inline src/dmd/expressionsem.d:1281]: Why is this deprecation inside a function named `resolveProperties` ?

**Pull request merged**